config_defaults.yml

```diff
@@ -1,29 +1,72 @@
 ---
 .travis.yml:
-  script: "\"bundle exec rake validate lint spec\""
+  supported_oss: &supported_oss
+    puppetlabs-acl:                      [windows]
+    puppetlabs-chocolatey:               [windows]
+    puppetlabs-dism:                     [windows]
+    puppetlabs-dsc:                      [windows]
+    puppetlabs-iis:                      [windows]
+    puppetlabs-powershell:               [windows]
+    puppetlabs-reboot:                   [windows]
+    puppetlabs-registry:                 [windows]
+    puppetlabs-sqlserver:                [windows]
+    puppetlabs-wsus_client:              [windows]
+    puppetlabs-inifile:                  [linux,windows]
+    puppetlabs-stdlib:                   [linux,windows]
+    puppetlabs-accounts:                 [linux]
+    puppetlabs-apache:                   [linux]
+    puppetlabs-apt:                      [linux]
+    puppetlabs-concat:                   [linux]
+    puppetlabs-firewall:                 [linux]
+    puppetlabs-haproxy:                  [linux]
+    puppetlabs-hocon:                    [linux]
+    puppetlabs-ibm_installation_manager: [linux]
+    puppetlabs-inifile:                  [linux]
+    puppetlabs-java:                     [linux]
+    puppetlabs-java_ks:                  [linux]
+    puppetlabs-mongodb:                  [linux]
+    puppetlabs-motd:                     [linux]
+    puppetlabs-mysql:                    [linux]
+    puppetlabs-ntp:                      [linux]
+    puppetlabs-postgresql:               [linux]
+    puppetlabs-rabbitmq:                 [linux]
+    puppetlabs-stdlib:                   [linux]
+    puppetlabs-tagmail:                  [linux]
+    puppetlabs-tftp:                     [linux]
+    puppetlabs-tomcat:                   [linux]
+    puppetlabs-vcsrepo:                  [linux]
+    puppetlabs-vsphere:                  [linux]
+    puppetlabs-xinetd:                   [linux]
+  script: "\"bundle exec rake release_checks\""
+  docker_sets:
+  - set: docker/ubuntu-14.04
+  - set: docker/centos-7
+  docker_defaults:
+    # values will replace @@SET@@ with the docker_sets' value
+    rvm: 2.3.1
+    sudo: required
+    dist: trusty
+    services: docker
+    env: PUPPET_INSTALL_TYPE=agent BEAKER_debug=true BEAKER_set=@@SET@@
+    script: bundle exec rake beaker
   includes:
-  - rvm: 1.9.3
-    env: PUPPET_GEM_VERSION="~> 3.0"
-  - rvm: 2.0.0
-    env: PUPPET_GEM_VERSION="~> 3.0"
-  - rvm: 2.1.5
-    env: PUPPET_GEM_VERSION="~> 4.0"
   - rvm: 2.3.1
     env: PUPPET_GEM_VERSION="~> 4.0"
+    bundler_args: --without system_tests
+  - rvm: 2.1.5
```

Comment:
The oldest ruby version we support in PE is 2.1.7 in PE 2015.2.3 (puppet 4.2.3). After March 31, 2017 the oldest supported ruby will be 2.1.9 running puppet 4.5.2 in PE 2016.2.1. 

After April 30, 2017 we down to supporting the PE 2016.4 LTS containing ruby 2.1.9, and puppet 4.7.0.

Maybe we should start explicitly specifying those version combinations in the cells here, to cover (and maintain) the right combinations. And have one for the latest version to avoid scrambling updates when we release new versions.